### PR TITLE
Adds tracking for common events

### LINKED ISSUE #270: [Analytics] Implement Client Side Analytics
## User Story

- [x] As a user viewing content, I should trigger an analytics event when I **view** a piece of content.
- [x] As a user viewing content, I should trigger an analytics event when I **Like** a piece of content.
- [x] As a user viewing content, I should trigger an analytics event when I **Share** a piece of content.
- [x] As a user, I should trigger an analytics event when I **Login**
- [x] As a user, I should trigger an analytics event when I **Signup**
- [x] As a user, I should trigger an analytics event when I **Forgot my Password**
- [x] As a user, I should trigger an analytics event when I **Play Audio/Video**
- [x] As a user, I should trigger an analytics event when I **Pause Audio/Video**
- [ ] As a user, I should trigger an analytics event when I **Follow a Link (?)**

Inspiration for events came from here: 
https://github.com/NewSpring/Apollos/blob/alpha/src/%40utils/analytics/index.js#L6

#### Problem Details _(Optional)_

* Screenshots, GIFs, or a video
* Any additional information (version, device, etc.)

#### Acceptance Criteria _(Optional)_

* [ ] It should [SPEC_DETAIL]

### PR DESCRIPTION
**Pull request by vinnyjth**
Fixes #270 

Adds tracking for the following events:

```
  LikeContent: 'Like Content',
  UnlikeContent: 'Unlike Content',
  ViewContent: 'View Content',
  ShareContent: 'Share Content',
  UserLogin: 'User Login',
  UserSignup: 'User Signup',
  UserLogout: 'UserLogout',
  UserForgotPassword: 'User Forgot Password',
  UserPlayedMedia: 'User Played Media',
```

This PR has the negative impact of decreasing test coverage in React Native. Anyone have any ideas on how to reduce that impact? Analytics testing is notoriously obnoxious ...
## Creator

- [ ] Build relevant tests if any apply
- [ ] Ensure there are no new warnings (in app AND in storybook)
- [ ] Upload an animated GIF showcasing the solution (if it applies)
- [ ] Set a relevant reviewer
- [ ] PR has a relevant title that adheres to our patch-notes/change-log style

## Reviewer

- [ ] Review CI build results
- [ ] Review function and form on iOS
- [ ] Review function and form on Android
- [ ] Review new stories if any apply
- [ ] Review test coverage: Are all new or changed states tested?
- [ ] Review new test logic
- [ ] Review that the PR title adheres to our patch-notes/change-log style

### CODE REVIEW

File: packages/apolloschurchapp/src/content-single/index.js [hunk 1]
```diff
@@ -151,6 +153,14 @@ class ContentSingle extends PureComponent {
                     colors: get(content, 'theme.colors'),
                   }}
                 >
+                  <TrackEventWhenLoaded
```
Comment (conrad-vanl) [inline packages/apolloschurchapp/src/content-single/index.js]: the `<Query>` component makes this sort of stuff super fun....

(versus say, a class-based component and `componentDidUpdate`)

File: packages/apolloschurchapp/src/analytics/TrackEventWhenLoaded.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { track } from './index';
+
+export default class TrackEventWhenLoaded extends PureComponent {
```
Comment (conrad-vanl) [inline packages/apolloschurchapp/src/analytics/TrackEventWhenLoaded.js]: We could add test coverage for this component - 

mock the `track` function, and assert that the `track` function is called after setting `loaded=true`. You can have a test case for `loaded=true` on mount, and a test case for rendering once with `loaded=false` then re-rendering with `loaded=true`

File: packages/apolloschurchapp/src/auth/LoginButton.js [hunk 3]
```diff
@@ -25,7 +27,16 @@ class LoginButton extends PureComponent {
 
   renderLogoutButton = () => (
     <Mutation mutation={logout}>
-      {(handleLogout) => <Button onPress={() => handleLogout()}>Logout</Button>}
+      {(handleLogout) => (
+        <Button
+          onPress={() => {
+            track({ eventName: events.UserLogout });
```
[the remark this answers is not in the document]
Comment (vinnyjth) [reply packages/apolloschurchapp/src/auth/LoginButton.js]: 👍 I'm always an advocate of moving events there when possible. I'll take a deeper look and see how many of them I can migrate.

File: packages/apolloschurchapp/src/content-single/index.js (hunk 1 above)
Comment (vinnyjth) [reply packages/apolloschurchapp/src/content-single/index.js]: Yah :/ reason #11 query components aren't always great.

File: packages/apolloschurchapp/src/analytics/TrackEventWhenLoaded.js (hunk 2 above)
Comment (vinnyjth) [reply packages/apolloschurchapp/src/analytics/TrackEventWhenLoaded.js]: 👍 I'll give it a shot.

Comment (conrad-vanl) [reply packages/apolloschurchapp/src/analytics/TrackEventWhenLoaded.js]: If it's difficult to mock the `track` function, you could always turn the `track` function into a prop on the component, and just default it to our internal track function, but pass a `jest.fn` mock function in tests :)

**Comment by vinnyjth** [comment]
@conrad-vanl Addressed your feedback, thanks much!  Ready for review again.

**Pull request merged**